zerver/webhooks/semaphore2/view.py

```diff
@@ -0,0 +1,180 @@
+* **Blocks**:
+{blocks}
+""".strip()
+
+PULL_REQUEST_TEMPLATE = """
+**{email}** created a pull request to branch **{branch_name}** of \
+[{organization_name}/{project_name}]({repository_url}):
+* **Pull Request**: [{pull_request_title}]({pull_request_url})
+* **Pipeline**:
+  - Name: {pipeline_name}
+  - Result: {pipeline_result}
+* **Blocks**:
+{blocks}
+""".strip()
+
+DEFAULT_TEMPLATE = """
+An activity occured on branch **{branch_name}** of \
+[{organization_name}/{project_name}]({repository_url}):
+* **Author**: {email}
+* **Pipeline**:
+  - Name: {pipeline_name}
+  - Result: {pipeline_result}
+* **Blocks**:
+{blocks}
+""".strip()
+
+TOPIC_TEMPLATE = "{project}/{branch}"
+
+@api_key_only_webhook_view('Semaphore')
+@has_request_variables
+def api_semaphore2_webhook(request: HttpRequest, user_profile: UserProfile,
+                           payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
+
+    organization_name = payload["organization"]["name"]
+    project_name = payload["project"]["name"]
+    repository_url = payload["repository"]["url"]
+    pipeline_name = payload["pipeline"]["name"]
+    pipeline_result = payload["pipeline"]["result"]
+    author_email = payload["revision"]["sender"]["email"]
+    branch_name = payload["revision"]["branch"]["name"]
+    blocks = get_blocks_data(payload["blocks"])
+
+    if payload["revision"]["pull_request"] is None:
+        event = "push"
+    else:
+        event = "pull_request"
+        pull_request_title = payload["revision"]["pull_request"]["name"]
+        pull_request_number = payload["revision"]["pull_request"]["number"]
+        pull_request_url = get_pr_url(pull_request_number, repository_url)
+
+    if event == "push":
+        commit_id = payload["revision"]["commit_sha"]
+        commit_message = payload["revision"]["commit_message"]
+        commit_url = get_commit_url(commit_id, repository_url)
+        content = PUSH_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            commit_hash = commit_id[:7],
+            commit_message = commit_message,
+            commit_url = commit_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    elif event == "pull_request":
+        content = PULL_REQUEST_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            pull_request_title = pull_request_title,
+            pull_request_url = pull_request_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    else:  # nocoverage
```

Comment:
It would still be useful to retain the comment here that we never get to this branch.

Reply:
Okay, I'll add that comment.